tree/dataframe/src/RDFSnapshotHelpers.cxx

```diff
@@ -930,3 +995,260 @@ ROOT::Internal::RDF::UntypedSnapshotRNTupleHelper::MakeNew(void *newName)
+   const ColumnNames_t &bnames, const RSnapshotOptions &options, std::vector<bool> &&isDefine,
+   ROOT::Detail::RDF::RLoopManager *outputLoopMgr, ROOT::Detail::RDF::RLoopManager *inputLoopMgr,
+   const std::vector<const std::type_info *> &colTypeIDs)
+   : fOptions(options), fInputLoopManager{inputLoopMgr}, fOutputLoopManager{outputLoopMgr}
+{
+   EnsureValidSnapshotTTreeOutput(fOptions, std::string(treename), std::string(filename));
+
+   TFile::TContext fileCtxt;
+   fOutputHandle = std::make_shared<SnapshotOutputWriter>(
+      TFile::Open(filename.data(), fOptions.fMode.c_str(), /*ftitle=*/"",
+                  ROOT::CompressionSettings(fOptions.fCompressionAlgorithm, fOptions.fCompressionLevel)),
+      nullptr);
+   if (!fOutputHandle->fFile)
+      throw std::runtime_error(std::string{"Snapshot: could not create output file "} + std::string{filename});
+
+   TDirectory *outputDir = fOutputHandle->fFile.get();
+   if (!dirname.empty()) {
+      fOutputHandle->fDirectoryName = dirname;
+      TString checkupdate = fOptions.fMode;
+      checkupdate.ToLower();
+      if (checkupdate == "update")
+         outputDir =
+            fOutputHandle->fFile->mkdir(std::string{dirname}.c_str(), "", true); // do not overwrite existing directory
+      else
+         outputDir = fOutputHandle->fFile->mkdir(std::string{dirname}.c_str());
+   }
+
+   fOutputHandle->fTree = std::make_unique<TTree>(std::string{treename}.c_str(), std::string{treename}.c_str(),
+                                                  fOptions.fSplitLevel, /*dir=*/outputDir);
+   fOutputHandle->fOutputLoopManager = fOutputLoopManager;
+   if (fOptions.fAutoFlush)
+      fOutputHandle->fTree->SetAutoFlush(fOptions.fAutoFlush);
+
+   auto outputBranchNames = ReplaceDotWithUnderscore(bnames);
+
+   fBranchData.reserve(vbnames.size());
+   for (unsigned int i = 0; i < vbnames.size(); ++i) {
+      fOutputHandle->RegisterBranch(outputBranchNames[i], 0);
+      fBranchData.emplace_back(vbnames[i], outputBranchNames[i], isDefine[i], colTypeIDs[i]);
+   }
+}
+
+// SnapshotHelperWithVariations::~SnapshotHelperWithVariations() = default;
+
+/// Register a new column as a variation of the column at `originalColumnIndex`, and clone its properties.
+/// If a nominal column is registered here, it is written without changes, but it means that it will be masked
+/// in case its selection cuts don't pass.
+/// \param slot Task ID for MT runs.
+/// \param columnIndex Index where the data of this column will be passed into the helper.
+/// \param originalColumnIndex If the column being registered is a variation of a "nominal" column, this designates the
+/// original.
+///         Properties such as name and output type are cloned from the original.
+/// \param variationName The variation that this column belongs to. If "nominal" is used, this column is considered as
+/// the original.
+void SnapshotHelperWithVariations::RegisterVariedColumn(unsigned int /*slot*/, unsigned int columnIndex,
```

Comment:
This and other `SnapshotHelperWithVariations` methods are defined in this implementation file within a `namespace ROOT::Internal::RDF {` scope. I believe it's better to define them outside, so that the explicit spelling `ROOT::Internal::RDF::SnapshotHelperWithVariations` must be used everywhere (like it's done in the constructor definition). Clearly `SnapshotOutputWriter` is exempt because it's defined directly within this file.